Description: Compute hardware_concurrency() using get_nprocs() on GLIBC systems
 All glibc-based systems have get_nprocs(), which can be used to
 implement hardware_concurrency().
 This patch fixes a build failure on Debian/kfreebsd.
 http://lists.debian.org/debian-bsd/2010/01/msg00036.html
Author: Petr Salinger <Petr.Salinger@seznam.cz>
Forwarded: https://svn.boost.org/trac/boost/ticket/6100

--- boost1.44-1.44.0.orig/libs/thread/src/pthread/thread.cpp
+++ boost1.44-1.44.0/libs/thread/src/pthread/thread.cpp
@@ -14,7 +14,7 @@
 #include <boost/thread/once.hpp>
 #include <boost/thread/tss.hpp>
 #include <boost/throw_exception.hpp>
-#ifdef __linux__
+#ifdef __GLIBC__
 #include <sys/sysinfo.h>
 #elif defined(__APPLE__) || defined(__FreeBSD__)
 #include <sys/types.h>
@@ -382,7 +382,7 @@
 #elif defined(BOOST_HAS_UNISTD_H) && defined(_SC_NPROCESSORS_ONLN)
         int const count=sysconf(_SC_NPROCESSORS_ONLN);
         return (count>0)?count:0;
-#elif defined(_GNU_SOURCE)
+#elif defined(__GLIBC__)
         return get_nprocs();
 #else
         return 0;
